### PR TITLE
feat(react-docs-page): surface env var error during development

### PR DESCRIPTION
**Pull request by thiskevinwang**
🎟️ [Asana Task]()
🔍 [Preview Link](https://react-components-git-{branch-slug}-hashicorp.vercel.app)

---

## Description

This causes `react-docs-page` api client code to throw an error in **development** about missing env vars.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/docs-page/content-api/index.ts [hunk 1]
```diff
@@ -7,11 +7,26 @@ const DEFAULT_HEADERS = {
...
 }
 
+// Courtesy helper for warning about missing env vars during development
+const checkEnvVarsInDev = () => {
+  if (process.env.NODE_ENV === 'development') {
+    if (!MKTG_CONTENT_API || !MKTG_CONTENT_API_TOKEN) {
+      const message = [
+        'You might be missing the following environment variables:',
+        '`MKTG_CONTENT_API`, `MKTG_CONTENT_API_TOKEN`',
+      ].join(' ')
```
Comment (BRKalow) [inline packages/docs-page/content-api/index.ts:17]:
```suggestion
      const message = [
        'Missing environment variables required to fetch remote content:',
        MKTG_CONTENT_API ? '\t- `MKTG_CONTENT_API`' : false, 
        MKTG_CONTENT_API_TOKEN ? '\t- `MKTG_CONTENT_API_TOKEN`' : false,
        'Reach out to #team-web-platform to get the proper values.'
      ].filter(Boolean).join('\n')
```

Comment (thiskevinwang) [reply packages/docs-page/content-api/index.ts:17]:
```
   _____       _                  
  / ____|     | |                 
 | |  __  __ _| | __ ___  ___   _ 
 | | |_ |/ _` | |/ _` \ \/ / | | |
 | |__| | (_| | | (_| |>  <| |_| |
  \_____|\__,_|_|\__,_/_/\_\\__, |
 |  _ \          (_)         __/ |
 | |_) |_ __ __ _ _ _ __    |___/ 
 |  _ <| '__/ _` | | '_ \         
 | |_) | | | (_| | | | | |        
 |____/|_|  \__,_|_|_| |_|        
```

File: packages/docs-page/content-api/index.ts [hunk 2]
```diff
@@ -7,11 +7,28 @@ const DEFAULT_HEADERS = {
...
 }
 
+// Courtesy helper for warning about missing env vars during development
+const checkEnvVarsInDev = () => {
+  if (process.env.NODE_ENV === 'development') {
+    if (!MKTG_CONTENT_API || !MKTG_CONTENT_API_TOKEN) {
+      const message = [
+        'Missing environment variables required to fetch remote content:',
+        MKTG_CONTENT_API ? '\t- `MKTG_CONTENT_API`' : false, 
+        MKTG_CONTENT_API_TOKEN ? '\t- `MKTG_CONTENT_API_TOKEN`' : false,
```
Comment (BRKalow) [inline packages/docs-page/content-api/index.ts:17]: Oops I ninja edited....
```suggestion
        !MKTG_CONTENT_API ? '  - `MKTG_CONTENT_API`' : false, 
        !MKTG_CONTENT_API_TOKEN ? '  - `MKTG_CONTENT_API_TOKEN`' : false,
```

**Review by BRKalow — APPROVED** [review]
📦

**Pull request merged**